src/createAnimatedComponent.tsx

```diff
@@ -246,8 +246,7 @@ export default function createAnimatedComponent(
   invariant(
     typeof Component !== 'function' ||
       (Component.prototype && Component.prototype.isReactComponent),
-    '`createAnimatedComponent` does not support stateless functional components; ' +
-      'use a class component instead.'
+    `Looks like you're passing a ${Component.name} functional component to 'createAnimatedComponent'.\n\n'createAnimatedComponent' relies on passing refs to work correctly. Wrap your functional component with React.forwardRef() or use a class component instead.`
```

Comment:
1. The official name is "function components" ([source](https://legacy.reactjs.org/docs/components-and-props.html#function-and-class-components)).
2. We use backticks (`) to quote identifiers and other names in error messages.
3. Instead of complaining, why don't we just wrap the input component with `React.forwardRef` on Reanimated side if needed?
```suggestion
    `Looks like you're passing a function component \`${Component.name}\` to \`createAnimatedComponent\` function which supports only class components. Please wrap your function component with \`React.forwardRef()\` or use a class component instead.`
```

Reply:
Sorry I don't have the capacity right now to do more research on this.

Animated API just throws a warning prompting the user to wrap function component with `forwardRef`. Maybe they had a reason to do so 🤔